app/routes/circulars/circulars.server.ts

```diff
@@ -0,0 +1,214 @@
+    )
+  }
+
+  /**
+   * Gets all available circulars
+   * @param searchString - a string that will be checked against subjects,
+   * bodies, and authors of saved circulars. A blank value returns all.
+   * @returns an array of CircularModel objects
+   */
+  async getCirculars(searchString?: string) {
+    let result = await this.#scanCircularsTable(searchString)
+    let items: CircularModel[] = result.Items
+    while (result.LastEvaluatedKey) {
+      result = await this.#scanCircularsTable(
+        searchString,
+        result.LastEvaluatedKey
+      )
+      items = [...items, ...result.Items]
+    }
+    return items.sort((a, b) => b.circularId - a.circularId)
+  }
+
+  async #scanCircularsTable(searchString?: string, lastEvaluatedKey?: any) {
+    const db = await tables()
+    const result = await db.circulars.scan({
+      ExclusiveStartKey: lastEvaluatedKey,
+      ExpressionAttributeValues: searchString
+        ? {
+            ':searchTerm': searchString,
+          }
+        : undefined,
+      FilterExpression: searchString
+        ? 'contains(subject, :searchTerm) OR contains(body, :searchTerm) OR contains(submitter, :searchTerm)'
+        : undefined,
+    })
+    return result
+  }
+
+  /**
+   * Get a specified Circular by Id.
+   *
+   * Throws an HTTP error if:
+   *  - the provided Id does not match any existing circulars
+   *
+   * @param id - the number Id (ex. 12345) of a specific Circular
+   * @returns a CircularModel object
+   */
+  async get(circularId: number): Promise<CircularModel> {
+    const db = await tables()
+    const result = await db.circulars.get({
+      circularId: circularId,
+    })
+    if (!result)
+      throw new Response('The requested circular does not exist', {
+        status: 404,
+      })
+    return result
+  }
+
+  /**
+   * Adds a new entry into the GCN Circulars table
+   *
+   * Throws an HTTP error if:
+   *  - The current user is not signed in, verified by the class's #sub and #groups properties
+   *  - The current user is not in the submitters group
+   *  - Body or Subject are blank
+   * @param body - main content of the Circular
+   * @param subject - the title/subject line of the Circular
+   */
+  async createNewCircular(body: string, subject: string) {
+    if (!this.#sub || !this.#groups)
+      throw new Response('User is not signed in', { status: 403 })
+    if (!this.#groups.includes('gcn.nasa.gov/circular-submitter'))
+      throw new Response('User is not in the submitters group', {
+        status: 403,
+      })
+    if (!body || !subject)
+      throw new Response('Subject and Body cannot be blank', { status: 400 })
+
+    const db = await tables()
+    const doc = db._doc as unknown as DynamoDBDocument
+
+    const tableName = db.name('circulars')
+    const indexTableName = db.name('auto_increment_metadata')
+
+    const submitter = formatAuthor({
+      name: this.#name,
+      affiliation: this.#affiliation,
+      email: this.#email ?? '',
```

Comment:
How can the email be undefined?

Reply:
In the case where the getter methods are only being used. If the user is not signed in, but just reading/searching for circulars

Reply:
I don't understand. The purpose of this function is to put a new circular into the database. How would email be undefined?

Reply:
I just updated the check, been a little since I looked at the actual function. This should be resolved in the latest commit